bin/handler-sensu-deregister.rb

```diff
@@ -4,12 +4,28 @@
 require 'sensu-handler'
 
 class Deregister < Sensu::Handler
+  option :invalidate,
+         description: 'Invalidate Client',
+         short: '-i',
+         long: '--invalidate',
+         default: false
+
+  option :invalidate_expire,
+         description: 'Invalidate Duration (seconds)',
+         short: '-d duration',
+         long: '--duration duration',
+         required: false
+
   def handle
     delete_sensu_client!
   end
 
   def delete_sensu_client!
-    response = api_request(:DELETE, '/clients/' + @event['client']['name']).code
+    if config[:invalidate] && config[:invalidate_expire]
+      response = api_request(:DELETE, '/clients/' + @event['client']['name'] + "?invalidate=#{config[:invalidate]}&#{config[:invalidate_expire]}").code
+    else
```

Comment:
this changes the default behavior (making it a breaking change) you can avoid this by adding another elsif with what you have below and an else with the old api request.